lib/srv/desktop/windows_server.go

```diff
@@ -1213,43 +1213,55 @@ func timer() func() int64 {
 	}
 }
 
+type entry struct {
+	sid               string
+	distinguishedName string
+}
+
 // generateUserCert generates a keypair for the given Windows username,
 // optionally querying LDAP for the user's Security Identifier.
 func (s *WindowsService) generateUserCert(ctx context.Context, username string, ttl time.Duration, desktop types.WindowsDesktop, createUsers bool, groups []string) (certDER, keyDER []byte, err error) {
 	var activeDirectorySID string
+	var distinguishedName string
+
 	if !desktop.NonAD() {
 		// Use FnCache to fetch the SID, or load it from cache if we already have it
 		// The cache key is the username and domain combined to handle multi-domain setups
 		cacheKey := fmt.Sprintf("%s@%s", username, desktop.GetDomain())
```

Comment:
The cache key would need to be based on the user's domain, not the desktop's domain, right?